tests/test_giticket.py

```diff
@@ -9,24 +9,100 @@
 from giticket.giticket import main
 from giticket.giticket import update_commit_message
 
-
 TESTING_MODULE = 'giticket.giticket'
 
 
-@pytest.mark.parametrize(
-    ('msg'),
-    (
-        'Test',
-        'JIRA-1234_Test',
-    ),
-)
+@pytest.mark.parametrize('msg', (
+    'Test ABC-1 message',
+    'ABC-2 Test message',
+    'Test message ABC-3',
+))
 @mock.patch(TESTING_MODULE + '.get_branch_name')
-def test_update_commit_message(mock_branch_name, msg, tmpdir):
+def test_update_commit_message_no_modification(mock_branch_name, msg, tmpdir):
     mock_branch_name.return_value = 'JIRA-1234_new_feature'
     path = tmpdir.join('file.txt')
     path.write(msg)
-    update_commit_message(six.text_type(path), '[A-Z]+-\d+', '{ticket} {commit_msg}')
-    assert 'JIRA-1234' in path.read()
```

Comment:
So this flow is now split and verified in `test_update_commit_message_no_modification` and `test_update_commit_message_underscore_split_mode`, I also made the check strict.